### ISSUE TITLE
cmd/objdump: order of CMP's arguments doesn't match go tool compile

### ISSUE THREAD

**Issue by eliben** [golang/go#60920]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.20.1 linux/amd64
</pre>

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/eliben/.cache/go-build"
GOENV="/home/eliben/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/eliben/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/eliben/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.20.1"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/eliben/eli/eliben-code/go/assembly/go.mod"
GOWORK=""
CGO_CFLAGS="-O2 -g"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-O2 -g"
CGO_FFLAGS="-O2 -g"
CGO_LDFLAGS="-O2 -g"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build2840069826=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?

Consider this Go code:

```
package assembly

func mycond(n int) int {
	if n > 15 {
		return 25
	} else {
		return 9
	}
}
```

I compile this code and emit assembly using two approaches:

1. `go tool compile -S`
2. `go build` followed by `go tool objdump`

### What did you expect to see?

The same order of arguments in CMP instructions

### What did you see instead?

In the output of `go tool compile -S`   I see:

```
CMPQ	AX, $15
JLE	<label for return 9>
```

In the output of `go tool objdump` I see:

```
CMPQ $0xf, AX		
JLE <label for return 9>
```

The order of arguments is different. In assembly input (for example https://cs.opensource.google/go/go/+/master:src/hash/crc32/crc32_amd64.s), the order matches what `go tool compile -S` produces:

	CMPQ CX, $8
	JL less_than_8

 it seems like the issue is with `go tool objdump`

**Comment by randall77** [comment]
The fix has been submitted to x/arch. I think at this point it is best to wait for 1.22 to vendor it into the stdlib.
(It has been broken ~forever, so no hurry.)

**Comment by dmitshur** [comment]
CL 505375 fixed this in x/arch and its footer said:

> Update [golang/go#60920](https://github.com/golang/go/issues/60920)
> (This fixes the underlying issue, but the actual "fixes" comment needs to go
on the CL that vendors x/arch containing this CL into the main branch.)

The fix was pulled into the main Go repo in [CL 509099](https://go.dev/c/509099), which was included in Go 1.21 starting with RC 3. So moving this to Go1.21 milestone.
